packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -335,7 +335,7 @@ export class FetchInstrumentation extends InstrumentationBase<
               return original
                 .apply(this, [url, options])
                 .then(
-                  onSuccess.bind(this, span, resolve),
+                  (onSuccess as any).bind(this, span, resolve),
```

Comment:
There might be a better solution then casting but I'm not an export on fetch and there are quite some indirections to follow.